R/rd-data.R

```diff
@@ -150,16 +151,73 @@ as_data.tag_examples <- function(x, ...,
 
   id_generator <- UniqueId$new()
 
-  html <- purrr::pmap_chr(
+  pieces <- purrr::pmap(
     list(code = code, run = run, show = show),
     format_example_chunk,
     env = child_env(env),
     topic = topic,
     obj_id = id_generator$id
   )
-  paste(html, collapse = "")
+
+  # flatten here to ensure that input and output blocks are at the same level.
+  pieces <- unname(purrr::flatten(pieces))
```

Comment:
Could also check for length == 2 lists, which are always input and output. I think there's at most one level, so flatten should be safe in this case. Inputs that are not replayed (e.g. from dontrun) are already flat.